src/compiler.rs

```diff
@@ -174,12 +174,16 @@ impl<CM: CellManager, SSB: StepSelectorBuilder> Compiler<CM, SSB> {
             panic!("Shared signals and fixed signals are not supported for circuits with different step heights. Using a different cell manager might fix this problem.");
         }
 
-        for forward_signal in sc.exposed.clone() {
-            let forward_placement = unit.placement.get_forward_placement(&forward_signal);
-            let exposed = (forward_placement.column, forward_placement.rotation);
-            unit.exposed.push(exposed);
+        if !unit.placement.same_height() {
+            panic!("Cannot calculate the number of rows");
         }
 
+        unit.num_rows = ast.num_steps * (unit.placement.first_step_height() as usize);
+
+        self.compile_fixed(ast, &mut unit);
+
+        self.compile_exposed(ast, &mut unit);
+
```

Comment:
We are limiting ALL circuits to those with the same step height for all step instances?